unstructured_ingest/processes/connectors/fsspec/fsspec.py

```diff
@@ -270,6 +270,22 @@ class FsspecDownloader(Downloader):
     download_config: Optional[FsspecDownloaderConfigT] = field(
         default_factory=lambda: FsspecDownloaderConfig()
     )
+    
+    def get_download_path(self, file_data: FileData) -> Optional[Path]:
+        if not file_data.source_identifiers:
+            return None
+        
+        filename = file_data.source_identifiers.filename
+        if not filename:
+            return None
```

Comment:
define both booleans as variables, join them with an `and` and return `None` once